### PR TITLE
[KIWI-1283] BAV test harness

### PR DESCRIPTION
**Pull request by Caitlin-cooling**
### What changed

Adds BAV test harness and api utils

### Why did it change

To enable automation testing of BAV 

### Screenshots

Send event to TxMA queue
Screenshot 2023-10-11 at 3 34 53 pm

Event shows up in test harness bucket
Screenshot 2023-10-11 at 3 35 09 pm

And can be fetched using API
Screenshot 2023-10-11 at 3 37 26 pm

Item in DB
Screenshot 2023-10-11 at 3 38 47 pm

Can be fetched using API
Screenshot 2023-10-11 at 3 41 20 pm

### Issue tracking
- [KIWI-1283](https://govukverify.atlassian.net/browse/KIWI-1283)

### PII logging

- [x] Verified that no PII data is being logged

### Environment variables or secrets
- [x] Documented in the [README](./blob/main/README.md)
- [x] Added to deployment repository
- [x] Added to local startup repository

### Other considerations

- [x] Update [README](./blob/main/README.md) with any new instructions or tasks

[KIWI-1283]: https://govukverify.atlassian.net/browse/KIWI-1283?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by Caitlin-cooling** [comment]
TODO: DNS and github workflows

File: test-harness/samconfig.toml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+version = 0.1
+[dev]
+[dev.deploy]
+[dev.deploy.parameters]
+stack_name = "bav-test-harness"
+s3_prefix = "bav-test-harness"
+region = "eu-west-2"
+confirm_changeset = false
+capabilities = "CAPABILITY_IAM CAPABILITY_NAMED_IAM"
+parameter_overrides = "CodeSigningConfigArn=\"none\" Environment=\"dev\" PermissionsBoundary=\"none\" SecretPrefix=\"none\" VpcStackName=\"vpc-cri\" BackendStack=\"bav-api\""
```
Comment (ViveakV) [inline test-harness/samconfig.toml:10]: bav-cri-api?

Comment (Caitlin-cooling) [reply test-harness/samconfig.toml]: yes, changesd

File: .github/workflows/pull-request-test-harness.yml [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+
+      - name: Setup SAM CLI
+        uses: aws-actions/setup-sam@v2
+        with:
+          use-installer: true
+
+      - name: Assume temporary AWS role
+        uses: aws-actions/configure-aws-credentials@v2
+        with:
+          role-to-assume: ${{ secrets.GH_DEV_TEST_HARNESS_VALIDATE_ROLE_ARN }}
```
Comment (ViveakV) [inline .github/workflows/pull-request-test-harness.yml:89]: GH_DEV_TEST_HARNESS_VALIDATE_ROLE_ARN is missing in Github Secrets

Comment (Caitlin-cooling) [reply .github/workflows/pull-request-test-harness.yml:89]: Raised ticket

**Review by ViveakV — DISMISSED** [review]

**Review by ViveakV — APPROVED** [review]

**Pull request merged**